src/sas/qtgui/Perspectives/Corfunc/CorfuncPerspective.py

```diff
@@ -263,16 +271,16 @@ def transform(self):
 
         method = "fourier"
 
-        extrap = self._q_space_plot.extrap
+        extrap = self.extrap
         background = float(self.model.item(WIDGETS.W_BACKGROUND).text())
 
         def updatefn(msg):
             """Report progress of transformation."""
             self.communicate.statusBarUpdateSignal.emit(msg)
 
-        def completefn(transforms):
+        def completefn(transformed_data: Tuple[Data1D, Data1D, Data1D]):
             """Extract the values from the transforms and plot"""
-            self.trigger.emit(transforms)
+            self.trigger.emit(TransformedData(*transformed_data)) # TODO: Make this return more structured data earlier
```

Comment:
`TransformedData` should probably have a constructor taking the tuple

Reply:
It's a namedtuple, really the line above it should be returning it, but all the Data1D issues prevent this.